File: Kap08/Java/Listing10/CallBehaviour.java
public class CallBehaviour {
    public static void main(String[] args){
        int iVal = 0;
        int[] iArVal = {0};
        incValue(iVal);
        incValue(iArVal);
        System.out.println("Ref 3: " + iVal);
        System.out.println("Ref 4: " + iArVal[0]);
      }
          
      public static void incValue(int iValPar) {
        iValPar++;
        System.out.println("Ref 1: " + iValPar);
      }
          
      public static void incValue(int[] iArValPar) {
        iArValPar [0]++;
        System.out.println("Ref 2: " + iArValPar [0]);
      }      
}

File: Kap10/JavaScript/Listing37/CallBehaviour.js
let iVal = 0;
let iArVal = [0];
incIVal(iVal);
incIArVal(iArVal);
console.log("Ref 3: " + iVal);
console.log("Ref 4: " + iArVal[0]);

function incIVal(iVal) {
    iVal++;
    console.log("Ref 1: " + iVal);
}
function incIArVal(iArVal) {
    iArVal[0]++;
    console.log("Ref 2: " + iArVal[0]);
}
